riscv-asm.md

```diff
@@ -202,8 +202,16 @@ Assembler Notation          | Description                    | Instruction / Mac
 %tls_ie_pcrel_hi(symbol) \* | TLS IE "Initial Exec" (HI20)   | auipc
 %tls_gd_pcrel_hi(symbol) \* | TLS GD "Global Dynamic" (HI20) | auipc
 %got_pcrel_hi(symbol) \*    | GOT PC-relative (HI20)         | auipc
+%gprel_hi(symbol)           | GP-relative (HI20)             | lui
+%gprel_lo(symbol)           | GP-relative (LO12)             | load, store, add
```

Comment:
What is the reason to skip `%gprel_add(symbol)` (and `%got_gprel_add(symbol)`)?

Reply:
That's what `%gprel(symbol)` and `%got_gprel(symbol)` are for as well.  They are overloaded for `add`, `la`, loads, stores.

Reply:
What if I want to write out those sequences longhand, like you can with the tprel sequences? Something like the following in the .S file:

```
  lui a0, %gprel_hi(symbol+off)
  add a0, a0, gp, %gprel_add(symbol+off)
  addi a0, a0, %gprel_lo(symbol+off)
```

Is this something we want to allow?

Reply:
Most definitely.

```
lui a0, %gprel_hi(symbol+off)
add a0, a0, gp, %gprel(symbol+off)
addi a0, a0, %gprel_lo(symbol+off)
```

Reply:
Ok, neat. My preference is for the former, because that mostly matches what has come before with the other relocation types.

Reply:
For the sake of uniformity, would you rather prefer `add a0, a0, gp, %gprel_add(symbol+off)` over `add a0, a0, gp, %gprel(symbol+off)`?

Reply:
Potentially, but maybe we can parse both anyway, and have `%gprel(symbol+off)` as canonical?

Reply:
Having `%gprel` be canonical is sad. I think it would be much better to have `%gprel(...)` be purely used in the pseudoinstructions, and maintain consistency otherwise by using `%gprel_add` to tag the `add` instruction. Overloading meaning adds needless mental overhead, whereas some semblance of consistency makes it much easier to move between models.

I personally don't like using `la rd, %gprel(symbol)` etc even, and would rather those be separate pseudoinstructions entirely. But that's a different discussion and more personal preference than actual concern.

Reply:
I don't have a problem with `%gprel_add`.